src/chunk.c

```diff
@@ -3375,48 +3399,85 @@ list_return_srf(FunctionCallInfo fcinfo)
 		SRF_RETURN_DONE(funcctx);
 }
 
-static void
-drop_remote_chunks(FunctionCallInfo fcinfo, const Name schema_name, const Name table_name,
-				   List *data_node_oids)
+/*
+ * Find either the hypertable or the materialized hypertable, if the relid is
+ * a continuous aggregate, for the relid.
+ *
+ * Will error if relid is not a hypertable or view (or cannot be found) or if
+ * it is a materialized hypertable.
+ */
+static Hypertable *
+find_hypertable_for_drop_by_relid(Cache *hcache, Oid relid)
 {
-	/* Wildcard drops not supported on distributed hypertables */
-	if (schema_name == NULL && table_name == NULL)
-		ereport(ERROR,
-				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-				 errmsg("cannot use wildcard to drop chunks on distributed hypertables"),
-				 errhint("Drop chunks on each distributed hypertable individually.")));
-
-	/* The schema name must be present when dropping remote chunks because the
-	 * search path on the connection is always set to pg_catalog. Thus, the
-	 * data node will not be able to resolve the same hypertables without the
-	 * schema. */
-	if (schema_name == NULL)
+	const char *rel_name;
+	Hypertable *ht;
+
+	rel_name = get_rel_name(relid);
+	if (!rel_name)
 		ereport(ERROR,
-				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-				 errmsg("schema required when dropping chunks on distributed hypertables")));
+				(errcode(ERRCODE_UNDEFINED_TABLE),
+				 errmsg("OID %u does not refer to a table or view", relid)));
 
-	/* Make sure the schema is set in case it was not given on the command line */
-	FC_ARG(fcinfo, 2) = NameGetDatum(schema_name);
-	FC_NULL(fcinfo, 2) = false;
+	ht = ts_hypertable_cache_get_entry(hcache, relid, CACHE_FLAG_MISSING_OK);
+	if (ht)
+	{
+		const ContinuousAggHypertableStatus status = ts_continuous_agg_hypertable_status(ht->fd.id);
+		if (status & HypertableIsMaterialization)
```

Comment:
It is not necessary since `ContinuousAggHypertableStatus` is a bitmask:

```
typedef enum ContinuousAggHypertableStatus
{
  HypertableIsNotContinuousAgg = 0,
  HypertableIsMaterialization = 1,
  HypertableIsRawTable = 2,
  HypertableIsMaterializationAndRaw = HypertableIsMaterialization | HypertableIsRawTable,
} ContinuousAggHypertableStatus;
```

Reply:
This code changes existing logic. A cagg table cannot be a raw + materialization now. But it is something we might support in the future.. We should continue to explicitly check for this in the code given that it is one of the valid return values for Status.

Reply:
Not sure what I am missing.

From the typedef, we have

- `HypertableIsNotContinuousAgg` = 0,
- `HypertableIsMaterialization` = 1,
- `HypertableIsRawTable` = 2,
- `HypertableIsMaterializationAndRaw` = `HypertableIsMaterialization | HypertableIsRawTable` = 3

Simplifying the code a little (to just reason about when to print an
error or not and just discussing the `status` variable, we have this
in the old code:

```c
switch (status)
{
	case HypertableIsMaterialization:
		/* error */
		break;
	case HypertableIsMaterializationAndRaw:
		/* error */
 		break;
 	case HypertableIsRawTable:
	    ... /* no error */
 	default:
	    ... /* no error */
}
```

Which gives:

- If `status = 0` then OK
- If `status = 1` then ERROR
- If `status = 2` then OK
- If `status = 3` then ERROR

And this in the new code:

```c
if (status & HypertableIsMaterialization)
	/* error */
```

Which gives:

- If `status = 0` then OK (since `status & 1 == 0`)
- If `status = 1` then ERROR (since `status & 1 != 0`)
- If `status = 2` then OK (since `status & 1 == 0`)
- If `status = 3` then ERROR (since `status & 1 != 0`)

These both produce the same ERROR/OK for the same values of `status`.

What am I missing?